src/Core/AbstractApi.php

```diff
@@ -167,7 +167,7 @@ private function stringifyBody($body): string
         return (string) $body;
     }
 
-    private function getSigner()
+    private function getSigner(?string $scope = null)
     {
         if (null === $this->signer) {
```

Comment:
the signer should vary with method's argument.

using the same AwsClient to perform 2 requests in 2 services will failed.